File: balanced_string.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        r_counter = 0
        l_counter = 0
        total_count = 0
        
        for i in s:
            if i == 'R':
                r_counter +=1
            else:
                l_counter += 1
            
            if r_counter == l_counter:
                total_count += 1
                r_counter = 0
                l_counter = 0
            
        return total_count
        

File: balanced_string.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int r_counter = 0;
        int l_counter = 0;
        int total_count = 0;
        
        for(int i = 0;i<s.length();i++){
            if(s[i]=='R'){
                r_counter++;
            }
            else{
                l_counter++;
            }
            if(r_counter == l_counter){
                total_count++;
            }
            }
        return total_count;
        }
        
    };
